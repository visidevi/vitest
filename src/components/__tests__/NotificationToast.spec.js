import { mount } from '@vue/test-utils'
import NotificationToast from '../NotificationToast.vue'
import { describe, it, expect } from 'vitest'

describe('NotificationToast', () => {
  it('renders the correct style for error', () => {
    const status = 'error'
    const wrapper = mount(NotificationToast, {
      props: { status, message: 'Hello' },
    })
    expect(wrapper.classes()).toContain('notification--error')
  })
  it('renders the correct style for success', () => {
    const message = ''
    const wrapper = mount(NotificationToast, {
      props: { message },
    })
    expect(wrapper.classes('notification--slide')).toBe(true)
  })
  it('renders the correct style for warning', async () => {
    const wrapper = mount(NotificationToast, {
      data() {
        return { clicked: true }
      },
    })
    const closeButton = wrapper.find('button')
    await closeButton.trigger('click')
    expect(wrapper.emitted()).toHaveProperty('clear-notification')
  })
  it('renders the correct message to viewer', () => {
    const status = 'info'
    const message = 'Something went wrong'
    const wrapper = mount(NotificationToast, {
      props: { status, message },
    })
    expect(wrapper.find('p').text()).toContain(message)
    expect(wrapper.classes()).toContain('notification--info')
  })
})
