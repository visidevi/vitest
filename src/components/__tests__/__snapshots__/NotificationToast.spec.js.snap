// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotificationToast > renders the correct style for error 1`] = `
"<div role="alert" class="notification notification--error notification--slide">
  <p class="notification__text">Hello</p><button title="close" class="notification__button"> âœ• </button>
</div>"
`;
